assets/js/modules/analytics/components/setup/SetupFormUA.js

```diff
@@ -16,24 +16,78 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { useMount } from 'react-use';
+
 /**
  * WordPress dependencies
  */
+import { __ } from '@wordpress/i18n';
 import { Fragment } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import { AccountSelect } from '../common';
+import Data from 'googlesitekit-data';
+import {
+	AccountSelect,
+	ExistingGTMPropertyNotice,
+	ExistingTagNotice,
+	ProfileSelect,
+	PropertySelect,
+	ProfileNameTextField,
+} from '../common';
+import { STORE_NAME, PROFILE_CREATE } from '../../datastore/constants';
+import { MODULES_ANALYTICS_4, PROPERTY_CREATE } from '../../../analytics-4/datastore/constants';
+import GA4PropertyNotice from '../common/GA4PropertyNotice';
+import StoreErrorNotices from '../../../../components/StoreErrorNotices';
+import { isValidPropertyID } from '../../util';
+
+const { useSelect, useDispatch } = Data;
 
 export default function SetupFormUA() {
+	const { selectProperty } = useDispatch( MODULES_ANALYTICS_4 );
+	const accounts = useSelect( ( select ) => select( STORE_NAME ).getAccounts() ) || [];
+	const hasExistingTag = useSelect( ( select ) => select( STORE_NAME ).hasExistingTag() );
+	const propertyID = useSelect( ( select ) => select( STORE_NAME ).getPropertyID() );
+
+	// Needed to conditionally show the profile name field and surrounding container.
+	const profileID = useSelect( ( select ) => select( STORE_NAME ).getProfileID() );
+
+	useMount( () => {
+		selectProperty( PROPERTY_CREATE );
+	} );
+
 	return (
 		<Fragment>
+			<StoreErrorNotices moduleSlug="analytics" storeName={ STORE_NAME } />
+
+			<ExistingTagNotice />
+			{ ! hasExistingTag && <ExistingGTMPropertyNotice /> }
+
+			{ ( !! accounts.length && ! hasExistingTag ) && (
+				<p className="googlesitekit-margin-bottom-0">
+					{ __( 'Please select the account information below. You can change this view later in your settings.', 'google-site-kit' ) }
+				</p>
+			) }
+
 			<div className="googlesitekit-setup-module__inputs">
 				<AccountSelect />
+
+				<PropertySelect />
+
+				{ isValidPropertyID( propertyID ) && <ProfileSelect /> }
```

Comment:
The `isValidPropertyID( propertyID )` check isn't needed here because the `ProfileSelect` component does the same internally: https://github.com/google/site-kit-wp/blob/6543b1cd989c03bfc62b22764b48a2e1e83464e5/assets/js/modules/analytics/components/common/ProfileSelect.js#L56-L58

Reply:
👍